### DISCUSSION TITLE
Math support for number fields

### DISCUSSION THREAD

**Discussion by dimaj** [firefly-iii/firefly-iii#6751] [General]
Hi!

I was wondering if it is possible to add simple math support to number input fields. 
Here is my use case:
I go to a store a buy a variety of things ranging from groceries (taxable and non-taxable), presents, medical supplies, etc. When I enter a new transaction for a said store, I am going to split the transaction so that each budget is accounted for.
It would be nice if I was able to enter things like: `4.99 + 3.99 + 2.99 * 1.1075 + 7.97 * 1.1075` and have the field auto-update (or update on save) to: `21.12`
In the example above, my local sales tax is 10.75%, hence `* 1.1075`.

The same logic should apply to the `budget` screen. If I get a monetary gift, i would like to add it to my personal spending budget and it would be nice if I could modify the exiting budget amount with the amount I have just received (ie: if current budget amount is 17.97 and I just got 28.53, I would like to add `+28.93` to the budget field and have it automatically update to: `46.9`)

Thanks!

**Comment by JC5** [comment]
I'm afraid not, for that you'll have to use your OS' calculator.

**Reply by dimaj** [reply]
I am just curious, is that due to unnecessary complications (similar to `I want the rules to do something they currently don't support.` section on https://docs.firefly-iii.org/firefly-iii/faq/other/) or some other reason?

**Reply by JC5** [reply]
A bit of both. It's complex and gives me a maintenance burden.

**Reply by dimaj** [reply]
That's fair.
Thanks.

**Reply by watertrainer** [reply]
would you be open to this if someone else implemented it? Especially in something like a split transaction i would really appreciate this feature, as i often know the total amount of the transaction and how much i want to split away, but not the remainder

**Reply by JC5** [reply]
No, sorry about that.
